### ISSUE TITLE
Slow runtime large images on CPU

### ISSUE THREAD

**Issue by DiTo97** [PaddlePaddle/PaddleOCR#8784]
- System environment: **Ubuntu 20.04**
- Version: **latest**
- Command code: -
- Complete error message: -

Hi @andyjpaddle, I am trying to use PaddleOCR to extract raw text from high resolution images (4k) of healthcare documents. The extraction quality is very satisfying, but the runtime it takes to get there is often over 16 seconds, which is out of scope for my intended use of the OCR engine.

Being images of healthcare documents, there is lots and lots of text, thus downscaling the images did not provide great results thus far, massively increasing the word error rate (WER).

I assume the issue might be the internal tokenizer of PaddleOCR which generates lots of visual tokens for large images, thus requiring much more time to complete.

Is there any idea that pops to your mind to mitigate the issue? Ideally, the raw text extraction should take around 5 seconds to enable the completion of further downstream tasks in a reasonable time.

For context: Python 3.7.15 on a single CPU

**Comment by tink2123** [comment]
In the reasoning process, you can try to turn on mkldnn to speed up the prediction process：
add args `--enable_mkldnn =True`

**Comment by DiTo97** [comment]
> In the reasoning process, you can try to turn on mkldnn to speed up the prediction process： add args `--enable_mkldnn =True`

How should I turn it on if I'm importing the PaddleOCR in the code, and not running the command-line scripts directly?

**Comment by fcarron** [comment]
Thanks for the suggestion with mkldnn.
Are there any disadvantages when using mkldnn? And are there other possibilities to make it even faster?

@DiTo97 
Here an example, this work for me:
```
ocr = PaddleOCR(
    use_angle_cls=True, lang="en", enable_mkldnn=True
) 
result = ocr.ocr(img_path, cls=True)
```

**Comment by DiTo97** [comment]
> Thanks for the suggestion with mkldnn. Are there any disadvantages when using mkldnn? And are there other possibilities to make it even faster?
> 
> @DiTo97 Here an example, this work for me:
> 
> ```
> ocr = PaddleOCR(
>     use_angle_cls=True, lang="en", enable_mkldnn=True
> ) 
> result = ocr.ocr(img_path, cls=True)
> ```

Thanks @fcarron for the example. I will experiment with it in the coming days.

I am also experimenting with downscaling the images. Hopefully, I will be able to find a good trade-off resolution between the WER of extractions and the runtime.

What about the *divide-et-impera* approach, @tink2123, i.e., cropping a large image in smaller patchers of reasonable size, processing them in parallel as a batch, and then merging the output words back together? Do you think it could speed up the process at all? Do you see any issues with the last merge step?

**Comment by DiTo97** [comment]
Just a quick update, @tink2123.

After more extensive testing with `enable_mlkdnn=True`, the runtime has gone down dramatically to very acceptable levels, so I think it's safe to close this issue for now. Thanks again for the support!

**Comment by jcyrss** [comment]
With the version 2.6.0.1 of PaddleOCR, with enable_mlkdnn enabled, the OCR result is way less accurate than that of the version 2.6.1.3.

The problem is, installing with command `pip install "paddleocr>=2.0.1" --upgrade PyMuPDF==1.21.1` ends up with the version 2.6.0.1 installed, instead of newer version. How so?

**Comment by ajinkya933** [comment]
> enable_mkldnn=True

Thanks this is really helpful
